service-base/src/main/java/org/eclipse/hono/service/http/HttpServiceBase.java

```diff
@@ -0,0 +1,461 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.service.http;
+
+import static java.net.HttpURLConnection.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.eclipse.hono.config.ServiceConfigProperties;
+import org.eclipse.hono.service.AbstractServiceBase;
+import org.eclipse.hono.util.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.http.*;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+
+/**
+ * A base class for implementing services using HTTP.
+ *
+ * @param <T> The type of configuration properties used by this service.
+ */
+public abstract class HttpServiceBase<T extends ServiceConfigProperties> extends AbstractServiceBase<T> {
+
+    /**
+     * The <em>application/json</em> content type.
+     */
+    protected static final String CONTENT_TYPE_JSON = "application/json";
+
+    /**
+     * The <em>application/json; charset=utf-8</em> content type.
+     */
+    protected static final String CONTENT_TYPE_JSON_UFT8 = "application/json; charset=utf-8";
+
+    /**
+     * Default file uploads directory used by Vert.x Web
+     */
+    protected static final String DEFAULT_UPLOADS_DIRECTORY = "/tmp";
+
+    private static final Logger LOG = LoggerFactory.getLogger(HttpServiceBase.class);
+
+    @Value("${spring.profiles.active:}")
+    private String activeProfiles;
+
+    private HttpServer server;
+    private HttpServer insecureServer;
+
+    /**
+     * Ends a response with HTTP status code 400 (Bad Request) and an optional message.
+     * <p>
+     * The content type of the message will be <em>text/plain</em>.
+     *
+     * @param response The HTTP response to write to.
+     * @param msg The message to write to the response's body (may be {@code null}).
+     * @throws NullPointerException if response is {@code null}.
+     */
+    protected static void badRequest(final HttpServerResponse response, final String msg) {
```

Comment:
Overriding static methods in Java is IMHO problematic (Background: https://stackoverflow.com/a/2831447/7740194). 
And the inspections of IntelliJ idea warn about methods declared `static` and `final` :-)